airbyte-webapp/src/pages/ConnectorBuilderPage/ConnectorBuilderPage.tsx

```diff
@@ -47,6 +50,11 @@ const ConnectorBuilderPageInner: React.FC = React.memo(() => {
         validationSchema={builderFormValidationSchema}
       >
         {(props) => {
+          if (showLandingPage) {
+            return (
+              <LandingPage hideLandingPage={hideLandingPage} switchToUI={switchToUI} switchToYaml={switchToYaml} />
```

Comment:
As this is part of the regular `ConnectorBuilderPage`, it initializes the form and test state services and tries to do a first fetch which fails with a 400.

I wonder whether we should split out the landing page into its own page with its own path - it seems easier to understand in the code (no "page in a page") and we can better control what side effects are triggered and avoid state leaking from one connector to another in case the user switches (also I think we won't get around that for having multiple builder projects anyway). 

This is what I could imagine as interim state:
* `connector-builder/edit` shows the actual connector builder UI (redirects to `connector-builder` on reset)
* `connector-builder` shows the landing page (for now redirects automatically to `connector-builder/edit` if there is a non-default state in localstorage)

I think we can have a separate service for the landing page wiring up with local storage (and later on with the APIs to save and load connectors).


Doesn't absolutely need to happen on this PR as this is a meaningful increment, but it seems like a better step towards the targeted feature scope - your call how to go with this.

Reply:
I like your suggestions here - I went ahead and added those changes now. I moved the local storage handling to its own ConnectorBuilderLocalStorage service/context, and moved the landing page into its own page / updated the routes and used redirects to go between.

I also found and fixed the issue with Reset All not always working properly - this was because it wasn't updating local storage properly, so I made sure it does that before navigating back to the landing page.